neoscore/western/tie.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from typing import Optional
+
+from neoscore.core.brush import BrushDef
+from neoscore.core.directions import DirectionY
+from neoscore.core.music_font import MusicFont
+from neoscore.core.music_path import MusicPath
+from neoscore.core.pen import PenDef
+from neoscore.core.point import PointDef
+from neoscore.core.positioned_object import PositionedObject
+from neoscore.core.spanner_2d import Spanner2D
+from neoscore.core.units import ZERO, Unit
+from neoscore.western.abstract_slur import AbstractSlur
+
+
+class Tie(AbstractSlur, MusicPath, Spanner2D):
+    """A tie.
+
+    While this is a path, it requires a music font from which to derive its appearance.
+    """
+
+    def __init__(
+        self,
+        pos: PointDef,
+        parent: PositionedObject,
+        end_x: Unit,
+        end_parent: Optional[PositionedObject] = None,
+        direction: DirectionY = DirectionY.UP,
+        height: Optional[Unit] = None,
+        arch_length: Optional[Unit] = None,
+        font: Optional[MusicFont] = None,
+        brush: Optional[BrushDef] = None,
+        pen: Optional[PenDef] = None,
+    ):
+        """
+        Args:
+            pos: The starting point.
+            parent: The parent for the starting position. If no font is provided,
+                this parent or one of its ancestors must implement :obj:`.HasStaffUnit`.
+            end_x: The stopping point.
+            end_parent: The parent for the ending position.
+                If ``None``, defaults to ``self``.
+            direction: The vertical direction the slur arches.
+            height: The ascent or descent of the curve given in absolute value.
+                If omitted, a reasonable default is derived from other properties.
+            arch_length: The x-offset of the outer ascent or descent curve control points.
+                Smaller values give tighter arches. If omitted, a reasonable default is
+                derived from other properties
+            font: If provided, this overrides any font found in the ancestor chain.
+            brush: The brush to fill shapes with.
+            pen: The pen to draw outlines with.
+        """
+        MusicPath.__init__(self, pos, parent, font, brush, pen)
+
+        end_pos = (end_x, ZERO)
+        Spanner2D.__init__(self, end_pos, end_parent or self)
```

Comment:
the tie really does need to be a proper 1-D spanner to guarantee horizontalness. It's only by luck that all chordrests use y=0 relative to the staff, so you can get away with setting end_pos.y = ZERO here. 1-D spanners are useful because they automatically work out the end_pos.y value needed to ensure the spanner is horizontal even when the start and end parents have different y-axis positions. You can see this bug in action with this little modification to your example:

```py
nh1 = Notehead(Mm(60), staff, "e", (1, 4))
nh2 = Notehead(Mm(80), staff, "b", (1, 4))
Tie(ORIGIN, nh1, ZERO, nh2)
```
![not_a_tie](https://user-images.githubusercontent.com/15976214/206886324-e742d41e-b40f-4b04-a459-f976c1b414b5.png)

Reply:
One fix, could be to only take in start parent with Tie.